octodns_transip/__init__.py

```diff
@@ -116,12 +201,40 @@ def populate(self, zone, target=False, lenient=False):
                         lenient=lenient,
                     )
                     zone.add_record(record, lenient=lenient)
+
         self.log.info(
             'populate:   found %s records', len(zone.records) - before
         )
 
         return True
 
+    def _process_desired_zone(self, desired):
+
+        for record in desired.records:
+            if record._type == 'NS' and record.name == '':
+
+                # Check values for FQDN, IP's are not supported
+                values = record.values if record.values else [record.value]
+
+                for value in values:
+
+                    if is_ipaddress(value.strip(".")):
+                        msg = f'ip address not supported for root NS value for {record.fqdn}'
+                        raise SupportsException(f'{self.id}: {msg}')
+
+                # Check if TTL differs
+                if record.ttl != self.ROOT_NS_TTL:
+                    record.ttl = self.ROOT_NS_TTL
```

Comment:
Point 3 in the `_process_desired_zone` guidelines calls out that the records should not be modified. 

https://github.com/octodns/octodns/blob/a1456cb1fcf00916ca06b204755834210a3ea9cf/octodns/provider/base.py#L35-L52

You can look around for examples, but generally you'll want to `record = record.copy()` and then modify the copy that's returned.

https://github.com/octodns/octodns/blob/a1456cb1fcf00916ca06b204755834210a3ea9cf/octodns/record/base.py#L273-L284

Reply:
Applied the record.copy function